### DISCUSSION TITLE
Prevent index from being overwritten

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#15530] [Q&A] [answered]
We have one multimap index that is created at startup, then we modify it from code to add more maps. Is it possible to prevent it from being overwritten when the app restarts?

**Comment by grisha-kotler** [comment]
You can set the `Lock Mode` for the index:
https://ravendb.net/docs/article-page/5.4/csharp/server/administration/index-administration#lock-mode

**Reply by marcuslindblom** [reply]
Ok, cool. Is it possible to apply the lock at creation time?

**Reply by marcuslindblom** [reply]
Is this the correct way to apply the lock?

`
IndexCreation.CreateIndexes(typeof(DefaultDocumentStoreHolder).Assembly, store); 
store.Maintenance.SendAsync(new SetIndexesLockOperation(index.Name, IndexLockMode.LockedIgnore)))
`

**Comment by grisha-kotler** [comment, marked as answer]
Yes.
You can also set on index creation.
`LockMode = IndexLockMode.LockedIgnore`.

**Reply by marcuslindblom** [reply]
As an extra parameter on `IndexCreation.CreateIndexes`?

**Reply by ayende** [reply]
No, in the index defintion itself.

**Reply by marcuslindblom** [reply]
Ok, thanks!
